### PR TITLE
add categories to CRDs

### PR DESCRIPTION
**Pull request by haraldatbmw**
### Description of your changes

To be able to get all managed resources of the provider-sql via `kubectl get managed` I extended all CRDs with the categories crossplane,managed,sql.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

I installed the CRDs in my local cluster and verified if `kubectl get managed` results in my sql-provider managed resources.

### CODE REVIEW

File: apis/postgresql/v1alpha1/provider_types.go
Comment (hasheddan) [inline apis/postgresql/v1alpha1/provider_types.go:61]: nitpick, we typically don't consider the `ProviderConfig` resources a managed resource. We have used category `provider` elsewhere (ref: https://github.com/crossplane/provider-aws/blob/211fada49e7d42cbed6035ef04e1e55ace941e27/apis/v1beta1/providerconfig_types.go#L50)
```suggestion
// +kubebuilder:resource:scope=Cluster,categories={crossplane,provider,sql}
```

File: apis/postgresql/v1alpha1/provider_types.go
Comment (hasheddan) [inline apis/postgresql/v1alpha1/provider_types.go:86]:
```suggestion
// +kubebuilder:resource:scope=Cluster,categories={crossplane,provider,sql}
```

**Review by srueg — APPROVED** [review]

**Comment by negz** [comment]
Awesome work, thank you @haraldatbmw!

**Comment by negz** [comment]
Looks like there's a commit missing a DCO sign-off. Could you follow the instructions at https://github.com/crossplane-contrib/provider-sql/pull/32/checks?check_run_id=2056837582 before I merge?

**Pull request merged**
